src/coreComponents/dataRepository/wrapperHelpers.hpp

```diff
@@ -490,6 +490,100 @@ pullDataFromConduitNode( Array< T, NDIM, PERMUTATION > & var,
   std::memcpy( var.data(), valuesNode.data_ptr(), numBytesFromArray );
 }
 
+
+
+template< typename T, typename INDEX_TYPE >
+std::enable_if_t< bufferOps::can_memcpy< T > >
+pushDataToConduitNode( ArrayOfArrays< T, INDEX_TYPE > const & var2,
+                       conduit::Node & node )
+{
+  ArrayOfArraysView< T const, INDEX_TYPE > const & var = var2.toViewConst();
+  internal::logOutputType( LvArray::system::demangleType( var ), "Output array via external pointer: " );
+
+  // ArrayOfArray::m_numArrays
+  INDEX_TYPE const numArrays = var.size();
+  conduit::DataType const numArraysType( conduitTypeInfo< INDEX_TYPE >::id, 1 );
+  node[ "__numberOfArrays__" ].set( numArraysType, const_cast< void * >( static_cast< void const * >(&numArrays) ) );
+
+  // ArrayOfArray::m_offsets
+  INDEX_TYPE const * const offsets = var.getOffsets();
+  conduit::DataType const offsetsType( conduitTypeInfo< INDEX_TYPE >::id, numArrays+1 );
+  node[ "__offsets__" ].set_external( offsetsType, const_cast< void * >( static_cast< void const * >( offsets ) ) );
+
+  // ArrayOfArray::m_sizes
+  INDEX_TYPE const * const sizes = var.getSizes();
+  conduit::DataType const sizesType( conduitTypeInfo< INDEX_TYPE >::id, numArrays );
+  node[ "__sizes__" ].set_external( sizesType, const_cast< void * >( static_cast< void const * >( sizes ) ) );
+
+  // Push the data into conduit
+  constexpr int conduitTypeID = conduitTypeInfo< T >::id;
+  constexpr int sizeofConduitType = conduitTypeInfo< T >::sizeOfConduitType;
+  conduit::DataType const dtype( conduitTypeID, offsets[numArrays] * sizeof( T ) / sizeofConduitType );
```

Comment:
I don't fully understand the size here. What' s in `offsets[numArrays]`? 

Shouldn't this be the size of each array * `numOfArrays` * `sizeof(T)` ?

Reply:
`offsets` is the location in the buffer that a subarray starts. The next `offset` is where the next one starts. So the last value `offsets[numArrays]` is the end of the buffers as there are `numArrays+1` values in `offsets`. So `offsets[numArrays] is the length of the buffer if I am not mistaken. @corbett5 @wrtobin ??

Reply:
oh I see. I would write this exact comment in the code.

Reply:
This sort of explanation should be in `ArrayOfArrays`.

Reply:
I agree with Randy. The last offset is one measure of the total length of the underlying memory. Other measures are the capacity from the underlying buffer, or offsets(numArrays - 1) + sizeOfArray(numArrays - 1) which would be the end of the valid values. For io you should only need to write up to this last value.

I'll give it a look over this evening if it hasn't been merged by then.

Reply:
@corbett5 I did notice it would be nice for a bit different nomenclature for this object....for example:

-  there is no accessor function that returns the actual size of the data buffer (offset[m_numArrays])
- Perhaps language of calling the "arrays" is confusing. I think "subArrays" would be better...however if we want to do an generic `NestedArrays` arrays object for an arbitrary nesting, it would be nice to come up with something extendible.
- The accessors for `m_offset` and `m_sizes` protected in `ArrayOfArraysView`. So that means that `ArrayOfArrays` can't provide public access. I am sure this is by design, but there needs to be more data helper functions to avoid needing direct access to these internal data arrays IMO.

Reply:
If you're accessing the underlying data (ie through `m_offsets` and `m_sizes`) you could be accessing uninitialized data. Which isn't a problem if you have an `ArrayOfArrays< double >`, but is very much a problem (or at least you need to know what you're doing) if you have an `ArrayOfArrays< string >`. So yeah I didn't make this easy on purpose.

There is a way to get the total underlying buffer capacity
```c++
  /**
   * @return Return the total number values that can be stored before reallocation.
   */
  LVARRAY_HOST_DEVICE constexpr inline
  INDEX_TYPE_NC valueCapacity() const
  { return m_values.capacity(); }
```

One way to do IO for the whole buffer would be

```c++
std::copy(arrayOfArryays[1].begin(),  arrayOfArrays[arrayOfArrays.size() - 1].end(), dst_iter);
```

Reply:
Certainly open for a name change. I thought I tried to call them inner arrays in the docs, but it's certainly confusing.